### ISSUE TITLE
Manual annotations removed by worker

### ISSUE THREAD

**Issue by anderoonies** [securingsincity/react-ace#483]
# Problem
I'm rendering a `react-ace` component with `annotations` like this:

```js
<ReactAce
  mode={'javascript'}
  // ...
  annotations={[{row: 0, column: 0, type: 'error', text: 'Some error.'}]}
/>
```

The manual annotation will show up briefly, then be removed once the worker runs.
This gif shows a few refreshes. On each page load the annotation shows up, but is then removed.
![annotation](https://user-images.githubusercontent.com/6645121/42171075-51116d50-7de6-11e8-815c-08adc2d721e4.gif)

I've disabled the worker with `setOptions={{useWorker: false}}`, but this also disables annotations done manually.

Thanks!

**Comment by charlesgiroux** [comment]
I feel like there should be 2 sets of annotations. The manually added annotations and the ones generated by the mode's parser. I think they currently collide and erase one another...

**Comment by anderoonies** [comment]
yeah, that's what's happening. i'm not sure if it's possible to have annotations exist outside of the lifecycle of the worker or not.

**Comment by mfalade** [comment]
I'm having the same problem. 
I want to be able to add custom annotations to the editor, while still preserving the annotations by the worker.

I have seen suggestions to disable the worker like so.
```javascript
setOptions={{useWorker: false}}
```

Doing this diables the worker's annotation however so all my linting/syntax highlighting is gone.

I want to be able to merge the worker's annotation with my custom annotation.

Can someone suggest how to go about this ?

**Comment by Ranjeet-Naidu** [comment]
Hi , I'm in a similar situation, unable to add custom error messages. Any solution?

**Comment by bkemper** [comment]
Instead of using the `annotations` prop, I manually `setAnnotations`.

```
const Editor = ({ annotations: customAnnotations = [] }) => {
  const [annotations, setAnnotations] = useState([]);
  const [editor, setEditor] = useState();

  const nextAnnotations = [
    ...annotations.filter(({ custom }) => !custom),  // annotations by worker
    ...customAnnotations.map((annotation) => ({ ...annotation, custom: true })) // flag for exclusion
  ];

  useEffect(() => {
    if (editor) {
      editor.getSession().setAnnotations(nextAnnotations);
    }
  }, [editor, JSON.stringify(nextAnnotations)]);

  return (
    <ReactAce
      mode="html"
      onLoad={setEditor}
      onValidate={setAnnotations}
      setOptions={{ useWorker: true }}
      value={props.value}
   />
}
```
